library/data/feature_flag/src/main/java/org/quiltmc/qsl/feature/flag/mixin/FeatureFlagBitSetMixin.java

```diff
@@ -0,0 +1,104 @@
+package org.quiltmc.qsl.feature.flag.mixin;
+
+import java.util.BitSet;
+import java.util.Collection;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+import net.minecraft.feature_flags.FeatureFlag;
+import net.minecraft.feature_flags.FeatureFlagBitSet;
+import net.minecraft.feature_flags.FeatureFlagGroup;
+
+import org.quiltmc.qsl.feature.flag.impl.QuiltFeatureFlagExtensions;
+
+@Mixin(FeatureFlagBitSet.class)
+public class FeatureFlagBitSetMixin {
+	@Unique
+	private final BitSet quilt$additionalFlags = new BitSet();
+
+	@Inject(method = "isIn", at = @At(value = "RETURN", ordinal = 2), cancellable = true)
```

Comment:
I'd rather see use of `@Slice` and `require =1` over ordinals. Ordinals are super fragile (what if the method adds an early return at the head?) and if this code ever breaks, it's not going to be clear to me (or whoever else takes up the responsibility of porting) what you were targeting. Feel free to ping me if you need help.

Reply:
In this case, ordinal is fairly safe because the function is 6 lines, an if-else chain with two early returns, then a simple bit mask check. If a slice is still recommended I can put one in, but this is one of the few cases where I think including a slice is unnecessary or possibly even more confusing.

Reply:
If possible, I'd like enough description of some kind (either a few words in a comment or a slice) that gives me enough of an idea that I could fix the ordinals on this mixin in an update without needing to open up an old QSL workspace and reverse-engineer the injection points myself.